gitlab-status/handler.go

```diff
@@ -32,8 +37,35 @@ func Handle(req []byte) string {
 		log.Printf("error while building base URL to the API: %s", urlErr.Error())
 	}
 
+	gatewayURL := os.Getenv("gateway_public_url")
+	if gatewayURL == "" {
+		log.Printf("empty gateway_public_url unable unable to construct logs and endpoint URLs")
+	}
+
+	logsURL, logsErr := FormatLogsURL(gatewayURL, &status.EventInfo)
+	if logsErr != nil {
+		log.Printf("error while formatting URL for logs: %s", logsErr.Error())
+	}
+
+	endpointURL, endpointErr := FormatEndpointURL(gatewayURL, &status.EventInfo)
+	if endpointErr != nil {
+		log.Printf("error while formatting endpoint URL: %s", endpointErr)
+	}
+
+	dashboardURL, dashboardErr := FormatDashboardURL(gatewayURL, &status.EventInfo)
+	if dashboardErr != nil {
+		log.Printf("the dashboard url failed: %s", dashboardErr.Error())
+	}
+
 	for _, commitStatus := range status.CommitStatuses {
-		reportErr := sendReport(url, token, commitStatus.Status, commitStatus.Description, commitStatus.Context)
```

Comment:
Can you please split the body of this for loop into a new function? sendCommitStatus or whatever makes sense.

Reply:
So move this block
```
		targetURL := endpointURL
		if commitStatus.Status != sdk.StatusSuccess {
			targetURL = logsURL
		}
		if commitStatus.Context == sdk.StackContext && commitStatus.Context != status.EventInfo.Service {
			targetURL = dashboardURL
		}
```
Into a new function?

Reply:
This logic ^ is now into its own method for convenience also added test